pkg/service/backup_scheduler.go

```diff
@@ -67,39 +59,37 @@ func NewScheduler(ctx context.Context) quartz.Scheduler {
 func scheduleRoutines(
 	scheduler Scheduler, routines map[string]*model.BackupRoutine, handlers BackupHandlerHolder,
 ) error {
-	jobStore.Lock()
-	defer jobStore.Unlock()
-	clear(jobStore.jobs)
-
+	newJobs := map[string]*quartz.JobDetail{}
 	var errs error
 	for routineName, routine := range routines {
 		if routine.Disabled {
 			continue
 		}
-		handler := handlers[routineName]
+		handler, _ := handlers.Load(routineName)
```

Comment:
Consider logging and continuing if not found, even if this scenario is not possible at this time.